libsql/internal/http/http.go

```diff
@@ -3,61 +3,90 @@ package http
 import (
 	"bytes"
 	"context"
+	"database/sql/driver"
 	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
+	"sort"
 	"time"
 
 	"github.com/xwb1989/sqlparser"
 )
 
 var httpClient = &http.Client{Timeout: 120 * time.Second}
 
-type params struct {
-	Names  []string
-	Values []any
-}
+type paramsType int
 
-func (p *params) MarshalJSON() ([]byte, error) {
-	if len(p.Values) == 0 {
-		return json.Marshal([]string{})
-	}
-	if len(p.Names) == 0 {
-		return json.Marshal(p.Values)
+const (
+	namedParameters paramsType = iota
+	positionalParameters
+)
+
+type parameters struct {
+	paramsValues []struct {
+		name  string
+		value any
 	}
-	m := map[string]interface{}{}
-	for idx := range p.Values {
-		m["@"+p.Names[idx]] = p.Values[idx]
+	paramsType paramsType
+}
+
+func (p *parameters) MarshalJSON() ([]byte, error) {
+	switch p.paramsType {
+	case namedParameters:
+		if len(p.paramsValues) == 0 {
+			return json.Marshal([]string{})
+		}
+		paramsMap := map[string]interface{}{}
+		for _, values := range p.paramsValues {
+			paramsMap[values.name] = values.value
```

Comment:
Have you tested this? Last time I checked sqld wasn't accepting params without @, : or $ prefix. I did some work in sqld to change this but I don't remember if I finished it

Reply:
I tested it with one of the examples in examples/sql/counter that uses named parameters with prefix @. As I suspected it failed:
```
LibSQL failed to bind provided query parameters: `value for parameter @country not found`
```

Reply:
Sorry, I forgot to talk about that. We have some limitations related to name parameters:

1. `sqlParser` only parses parameters if `:`. So we only accept parameters like ":parameterName" for now/
2. I've changed the way we need to pass name parameters. Previously, the driver automatically added `:` prefix for name parameters, but, since sqld accepts `@` and `$` prefix, I decided to don't add `:` prefix automatically and force the users to do so. So you need would need to call sql.Named("@country", "PL") after first point is fixed. You can test using `:` prefix by now to check that it works (I've tested already)

Reply:
You can't do (2) with databases/sql. It allows only param names with no prefix.

Reply:
I guess accepting `:name` is better than not accepting anything but it is not good that sqlParser does not support parameters names other than `:name`. I'm not sure if this isn't a blocker for using sqlParser.

Reply:
> You can't do (2) with databases/sql. It allows only param names with no prefix.

You're right. I remembered that I planned to do you I told you, but this exactly problem stopped me.

Now, after using ANTLR4 parser, it works fine with all prefixes (`:`, `$`, `@`). Also, I've just tested (using counter/main.go) http and libsql connections and both worked with named parameters